core/io/resource_loader.cpp

```diff
@@ -112,10 +112,21 @@ String ResourceFormatLoader::get_resource_script_class(const String &p_path) con
 
 ResourceUID::ID ResourceFormatLoader::get_resource_uid(const String &p_path) const {
 	int64_t uid = ResourceUID::INVALID_ID;
-	GDVIRTUAL_CALL(_get_resource_uid, p_path, uid);
+	if (!GDVIRTUAL_CALL(_get_resource_uid, p_path, uid)) {
+		if (!has_custom_uid_support()) {
+			Ref<FileAccess> file = FileAccess::open(p_path + ".uid", FileAccess::READ);
+			if (file.is_valid()) {
+				uid = ResourceUID::get_singleton()->text_to_id(file->get_line());
+			}
+		}
+	}
```

Comment:
You are checking twice for UID support. This can be simplified to
```suggestion
	if (has_custom_uid_support()) {
		GDVIRTUAL_CALL(_get_resource_uid, p_path, uid);
	} else {
		Ref<FileAccess> file = FileAccess::open(p_path + ".uid", FileAccess::READ);
		if (file.is_valid()) {
			uid = ResourceUID::get_singleton()->text_to_id(file->get_line());
		}
	}
```